genson/src/main/java/com/owlike/genson/reflect/VisibilityFilter.java

```diff
@@ -70,6 +70,14 @@ public VisibilityFilter(int... modifier) {
    * @return true if this member is visible according to this filter.
    */
   public final boolean isVisible(Member member) {
+    Class<?> clazz = member.getDeclaringClass();
+    String className = clazz.getName();
+    if(className.startsWith("java.") || className.startsWith("javax.")){
+      if(!Modifier.isPublic(member.getModifiers())){
+        return false;
+      }
+    }
+
     return isVisible(member.getModifiers());
```

Comment:
You forgot to include what I mentioned in the other PR, you want to leverage the same filter as isVisible. Otherwise this will include all fields/methods that are public. Some might be static, transient, etc, these shouldn't be included.

Reply:
This code is slightly different from the other PR.  The main difference being the `!` in front of Modifiers.isPublic.  In this PR, any non-public field in a base class is excluded. Otherwise, we perform the additional call to isVisible (line 81)

The previous PR was automatically including all public fields in base classes without considering the additional filters.  I can see how this might not be the clearest code however. I will modify the PR to be a bit more clear

Reply:
Oops my bad! Thanks for the explanation.